cmd/plakar/utils/locate_test.go

```diff
@@ -0,0 +1,155 @@
+
+	"github.com/PlakarKorp/plakar/repository"
+	"github.com/PlakarKorp/plakar/snapshot"
+	ptesting "github.com/PlakarKorp/plakar/testing"
+	"github.com/stretchr/testify/require"
+)
+
+func init() {
+	os.Setenv("TZ", "UTC")
+}
+
+func generateSnapshot(t *testing.T, bufOut *bytes.Buffer, bufErr *bytes.Buffer) (*repository.Repository, *snapshot.Snapshot) {
+	repo := ptesting.GenerateRepository(t, bufOut, bufErr, nil)
+	snap := ptesting.GenerateSnapshot(t, repo, []ptesting.MockFile{
+		ptesting.NewMockDir("subdir"),
+		ptesting.NewMockDir("another_subdir"),
+		ptesting.NewMockFile("subdir/dummy.txt", 0644, "hello dummy"),
+		ptesting.NewMockFile("subdir/foo.txt", 0644, "hello foo"),
+		ptesting.NewMockFile("subdir/to_exclude", 0644, "*/subdir/to_exclude\n"),
+		ptesting.NewMockFile("another_subdir/bar.txt", 0644, "hello bar"),
+	})
+	return repo, snap
+}
+
+func generateSnapshotWithMetadata(t *testing.T, repo *repository.Repository, opts ptesting.TestingOptions) *snapshot.Snapshot {
+	snap := ptesting.GenerateSnapshot(t, repo, []ptesting.MockFile{
+		ptesting.NewMockDir("subdir"),
+		ptesting.NewMockFile("subdir/dummy.txt", 0644, "hello dummy"),
+	}, opts)
+	require.NotNil(t, snap)
+	return snap
+}
+
+func TestParseSnapshotPath(t *testing.T) {
+	// Test case: Snapshot path with prefix and pattern
+	prefix, pattern := ParseSnapshotPath("prefix:pattern")
+	require.Equal(t, "prefix", prefix)
+	require.Equal(t, "pattern", pattern)
+
+	// Test case: Absolute path
+	prefix, pattern = ParseSnapshotPath("/absolute/path")
+	require.Equal(t, "", prefix)
+	require.Equal(t, "/absolute/path", pattern)
+
+	// Test case: Only prefix without pattern
+	prefix, pattern = ParseSnapshotPath("prefix")
+	require.Equal(t, "prefix", prefix)
+	require.Equal(t, "", pattern)
+
+	// Test case: Empty input
+	prefix, pattern = ParseSnapshotPath("")
+	require.Equal(t, "", prefix)
+	require.Equal(t, "", pattern)
+}
+
+func TestLookupSnapshotByPrefix(t *testing.T) {
+	bufOut := bytes.NewBuffer(nil)
+	bufErr := bytes.NewBuffer(nil)
+
+	repo, snap := generateSnapshot(t, bufOut, bufErr)
+	defer snap.Close()
+
+	// Test case: Prefix that matches a single snapshot
+	results := LookupSnapshotByPrefix(repo, hex.EncodeToString(snap.Header.GetIndexShortID()))
+	require.Len(t, results, 1)
+	require.Equal(t, results[0], snap.Header.Identifier)
+
+	// Test case: Prefix that matches no snapshots
+	results = LookupSnapshotByPrefix(repo, hex.EncodeToString([]byte{0x00}))
+	require.Len(t, results, 0)
+}
+
+func TestLocateSnapshotByPrefix(t *testing.T) {
+	bufOut := bytes.NewBuffer(nil)
+	bufErr := bytes.NewBuffer(nil)
+
+	repo, snap := generateSnapshot(t, bufOut, bufErr)
+	defer snap.Close()
+
+	found, err := LocateSnapshotByPrefix(repo, hex.EncodeToString(snap.Header.GetIndexShortID()))
+	require.NoError(t, err)
+	require.Equal(t, found, snap.Header.Identifier)
+
+	_, err = LocateSnapshotByPrefix(repo, hex.EncodeToString([]byte{0x00}))
+	require.EqualError(t, err, "no snapshot has prefix: 00")
```

Comment:
i fear that this might break from time to time if we're lucky enough to generate a snapshot that starts with 0x00.  Since we generate only one snapshot, we could do something like `snap.Header.Identifier[0] + 1` to get back a byte that's for sure not a prefix for anything.  (that sum should implicitly wrap around for 0xff to 0x00).

Reply:
good catch @omar-polo , updated in https://github.com/PlakarKorp/plakar/pull/842/commits/59ca7a82608ced3b5f97fe78583f637c3cf65dd6

Reply:
thank you!